tests/rptest/tests/tx_coordinator_migration_test.py

```diff
@@ -0,0 +1,171 @@
+
+import random
+from rptest.services.failure_injector import FailureInjector, FailureSpec
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.admin import Admin
+from rptest.services.redpanda import RESTART_LOG_ALLOW_LIST
+import confluent_kafka as ck
+from rptest.services.admin import Admin
+from rptest.utils.mode_checks import skip_debug_mode
+from ducktape.mark import parametrize
+
+
+class TxCoordinatorMigrationTest(RedpandaTest):
+    def __init__(self, test_context):
+        self.partition_count = 8
+        self.initial_tx_manager_partitions = 4
+        self.producer_count = 100
+        super(TxCoordinatorMigrationTest,
+              self).__init__(test_context=test_context,
+                             num_brokers=3,
+                             extra_rp_conf={
+                                 'transaction_coordinator_partitions':
+                                 self.initial_tx_manager_partitions
+                             })
+
+    def _tx_id(self, idx):
+        return f"test-producer-{idx}"
+
+    def _populate_tx_coordinator(self, topic):
+        def delivery_callback(err, msg):
+            if err:
+                assert False, "failed to deliver message: %s" % err
+
+        for i in range(self.producer_count):
+            producer = ck.Producer({
+                'bootstrap.servers': self.redpanda.brokers(),
+                'transactional.id': self._tx_id(i),
+                'transaction.timeout.ms': 10000,
+            })
+            producer.init_transactions()
+            producer.begin_transaction()
+            for m in range(random.randint(1, 50)):
+                producer.produce(topic,
+                                 f"p-{i}-key-{m}",
+                                 f"p-{i}-value-{m}",
+                                 random.randint(0, self.partition_count - 1),
+                                 callback=delivery_callback)
+            producer.commit_transaction()
+            producer.flush()
+
+    def _get_tx_id_mapping(self):
+        mapping = {}
+        admin = Admin(self.redpanda)
+        for idx in range(self.producer_count):
+            c = admin.find_tx_coordinator(self._tx_id(idx))
+            mapping[self._tx_id(
+                idx)] = f"{c['ntp']['topic']}/{c['ntp']['partition']}"
+
+        return mapping
+
+    def _get_tx_manager_topic_meta(self):
+        """_summary_
+            Collects partition metadata i.e. mapping between partition id and 
+            raft group, this mapping changes when topic is deleted and recreated
+        
+        """
+        metadata = Admin(self.redpanda).get_partitions(
+            namespace="kafka_internal", topic="tx")
+
+        return {(p['partition_id'], p['raft_group_id']) for p in metadata}
+
+    def _migrate_until_success(self):
+        admin = Admin(self.redpanda)
+        finished = False
+        cnt = 0
+        max_failures = 5
+        fi = FailureInjector(self.redpanda)
```

Comment:
do we need to use `with` with FailureInjector for cleanup?

Reply:
in general yes, but here we only inject `SIGKILL` so there is nothing to clean